### DISCUSSION TITLE
Creating a new driver

### DISCUSSION THREAD

**Discussion by AutoDMC** [ryangjchandler/orbit#90] [Q&A]
Hello:

I'm creating a new driver that doesn't extend Orbit\Drivers\FileDriver

I installed the version of Orbit that happens when I use `composer require ryangjchandler/orbit` (https://packagist.org/packages/ryangjchandler/orbit#1.x-dev)

However, there doesn't seem to be a Orbit\Contracts\Driver interface; I get "Undefined class 'Driver'" from PHPStorm

I'm creating the class under app/OrbitDrivers/SpecialDriver.php in my Laravel application.

I'll admit to being pretty new to Laravel, so I'm probably doing something stupid.  But I could use some pointers as to what I'm doing wrong.

```php
<?php

namespace App\OrbitDrivers;

use Orbit\Contracts\Driver as DriverContract;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Collection;

class SpecialDriver implements DriverContract
{
    public function shouldRestoreCache(string $directory): bool {
        return true;
    }

    public function save(Model $model, string $directory): bool {
        return false;
    }

    public function delete(Model $model, string $directory): bool {
        return false;
    }

    public function all(string $directory): Collection {
        $collection = Collection::make();

        return $collection;
    }
}
```

**Comment by ryangjchandler** [comment]
In your composer.json file, what version is against this package?

**Reply by AutoDMC** [reply]
`"ryangjchandler/orbit": "^1.0"`

**Reply by ryangjchandler** [reply]
Can you trying pinning the version to `1.0.1` to see if that resolves the problem? The original post you linked to `1.x-dev` which is an old tag that no longer exists.
